common/inbus/dto.py

```diff
@@ -0,0 +1,160 @@
+@dataclass
+class ConcreteActivity:
+    """
+    Concrete activity in schedule.
+    """
+
+    concreteActivityId: int # ID rozvrhové aktivity
+    template: str # activityTemplate např. P, P1, C, C1 ...
+    order: int # pořadí concreteActivity v rámci activity template, dohromady s activityTemplate pak dává např. P/01, P/02, P1/01, C/01 ...
+    subjectVersionId: int # ID verze předmětu
+    subjectVersionCompleteCode: str # kód verze předmětu (včetně čísla předmětu)
+    subjectId: int # ID předmětu
+    subjectAbbrev: str # zkratka předmětu
+    subjectTitle: str # název předmětu
+    educationTypeId: int # ID typu výuky (přednáška, cvičení, konzultace)
+    educationTypeAbbrev: str # zkratka typu výuky
+    educationTypeTitle: str # název typu výuky
+    semesterId: int # ID semestru
+    semesterTypeId: int # ID typu semestru
+    semesterTypeAbbrev: str # zkratka typy semestru
+    semesterTypeTitle: str # název typu semestru
+    academicYearId: int # ID akademického roku
+    academicYearTitle: str # název akademického roku
+    tutorialCentreId: int # ID konzultačního střediska
+    tutorialCentreAbbrev: str # zkratka konzultačního střediska
+    tutorialCentreTitle: str # název konzultačního střediska
+    educationWeekId: int # ID typu týdne výuky (každý, lichý, sudý, nepravidelně)
+    educationWeekTitle: str # název typu týdne výuky
+    beginScheduleWindowId: int # ID počáteční výukové jednotky (rozvrhového okna)
+    activityDuration: int # délka výuky - počet výukových jednotek následujících po sobě
+    beginTime: str # ($date-time) čas začátku výuky (HH:mm:ss)
+    endTime: str # ($date-time) čas konce výuky (HH:mm:ss)
+    weekDayId: int # ID dne v týdnu
+    weekDayAbbrev: str # zkratka dne v týdnu
+    weekDayTitle: str # název dne v týdnu
+    weekActivities: List[WeekActivity] # seznam výskytů aktivity v konkrétních týdnech výuky (využití při nepravidelné výuce), obsahuje: weekActivityId (Integer), weekNumber (Integer), date (Date)
+    roomFullcodes: str # optimalizační předpočítávaný atribut z vazby na místnosti
+    roomIds: List[int] # seznam ID místností
+    teacherIds: List[int] # seznam ID vyučujících (osob)
+    studyGroupIds: List[int] # seznam ID studijních skupin (má význam při rozvrhování na studijní skupiny)
+
+    # Default values are defined here, since INBUS API sometimes does not provide there values
+    # and Python need default values at the end of the definition
+    studyGroupCodes: str = '' # optimalizační předpočítávaný atribut z vazby na studijní skupiny
+    teacherFullNames: str = '' # optimalizační předpočítávaný atribut z vazby na vyučující
+    teacherLogins: str = '' # optimalizační předpočítávaný atribut z vazby na vyučující
+    teacherShortNames: str = '' # optimalizační předpočítávaný atribut z vazby na vyučující
+
+
+@serde.serde
+@dataclass
+class StudyRelation:
+    """
+    Relation of student and university.
+    """
+
+    studyRelationId: int # ID studijního vztahu
```

Comment:
Maybe we should list only the attributes that we care about? To make the code more readable and also to limit the number of fields that will break deserialization once they remove them :)

Reply:
Well, you're right. I think this is also used by Edison, so there's low probability of changing/breaking. Also, I'd like to separate this INBUS implementation into a completely different package and use it for broader use cases and this is "kind of" testbed :)